streamalert/artifact_extractor/artifact_extractor.py

```diff
@@ -42,11 +42,12 @@ def __init__(self, source_type, normalized_type, value, **kwargs):
                 record processed by classifier. This field is very useful for cross reference back
                 to the original record in the future.
         """
-        self._function = kwargs.get('function', 'not_specified')
-        self._record_id = kwargs.get('record_id', 'RESERVED')
-        self._source_type = source_type
-        self._type = normalized_type
-        self._value = value
+        # Enforce all fields are strings in a Artifact to prevent type corruption in Parquet format
+        self._function = str(kwargs.get('function', 'not_specified'))
+        self._record_id = str(kwargs.get('record_id', 'RESERVED'))
```

Comment:
can this `RESERVED` string literal be moved to a class constant instead and referenced like `cls.RESERVED` or `self.RESERVED`? and add a comment as to its purpose? I'm not a big fan of string literals just floating around

Reply:
👍 will make the change!

Reply:
It's meant to be temporary. Honestly I would just make it `None` and put a `FIXME`/`TODO` there

Reply:
I have comment about `record_id at line 41 https://github.com/airbnb/streamalert/blob/ebba5c51db122be5747def5fb4056076f01dd050/streamalert/artifact_extractor/artifact_extractor.py#L41-L43

Reply:
I noticed, but it doesn't really help when it's not near the actual thing it relates to. putting this as a class constant gives you the ability to at a relative comment right next to it